File: apps/dashboard/gonvex/system.go
package gonvextest

import (
	"time"

	"github.com/gonvex/gonvex/pkg/gonvex"
)

// HeartbeatResult is the (empty) payload returned by the demo heartbeat job.
type HeartbeatResult struct {
	OK bool `json:"ok"`
}

// RegisterSystem wires a small recurring job so the dashboard health overview
// shows live scheduler activity (running/queued/lag, throughput) out of the
// box. Remove or replace it with real background work for your project.
func RegisterSystem(app *gonvex.App) {
	app.InternalReducer("system.heartbeat", Heartbeat)
	app.Cron("heartbeat", 15*time.Second, "system.heartbeat", nil)
}

// Heartbeat is a no-op internal Reducer run on a schedule by the cron above.
func Heartbeat(ctx *gonvex.ReducerCtx, args struct{}) (HeartbeatResult, error) {
	return HeartbeatResult{OK: true}, nil
}

File: apps/dashboard/gonvex/system.ts
import { cron, internalReducer, schema } from "@gonvex/module-sdk";

export const heartbeat = internalReducer({
  args: schema.object({}),
  result: schema.object({ ok: schema.boolean() }),
  run: async () => ({ ok: true }),
});

export const heartbeatSchedule = cron({
  name: "heartbeat",
  function: "system.heartbeat",
  intervalMs: 15_000,
});
